idaes/core/solvers/tests/test_solvers.py

```diff
@@ -17,6 +17,11 @@
 from idaes.core.solvers import ipopt_has_linear_solver
 from idaes.core.solvers import petsc
 
+@pytest.mark.unit
+def test_petsc_available():
+    if not pyo.SolverFactory('petsc_snes').available():
+        raise Exception("Could not find petsc (petsc is an optional extra).")
```

Comment:
Minor comment, but is there a better exception we could use here (RuntimeError maybe)?